### ISSUE TITLE
Move comment-only rules from physical to token checker

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#5979]
E.g., anything in the:

```rust
while commented_lines_iter
    .next_if(|comment_range| line.range().contains_range(**comment_range))
    .is_some()
{
  ...
}
```

block of `crates/ruff/src/checkers/physical_lines.rs` can probably be moved to a token-based rule, which should be simpler (and more robust, actually, since those rules right now could erroneously match on comment-like syntax within strings).
